### PR TITLE
Adds directional swipe, makes swipe speed configurable through duration and fix iOS scroll

### PR DESCRIPTION
**Pull request by amanjeetsingh150**
### Adding directional swipe

Adds directional swipe: RIGHT, LEFT, DOWN, UP. This will help in use cases like swiping pagers in onboarding irrespective of device width height. API:
```
- swipe:
      direction: LEFT | RIGHT | UP | DOWN
```
PS: start and end coordinates still remain but are optional now.

### Controlling swipe speed
Swipe speed can be controlled by supplying duration. More the duration slower the swipe speed. The lesser the duration the more swipe speed. API:

```
- swipe:
     direction: LEFT
     duration: 300
```

### Fix iOS scrolling
Accounting scroll delta and duration which fixes iOS vertical scrolling as well

## Testing

https://user-images.githubusercontent.com/12881364/200290424-32c19707-3bbf-4eef-8658-7dfd175e947a.mp4

https://user-images.githubusercontent.com/12881364/200289793-119f3a61-56f5-4d06-b8d5-ab98b445bbb9.mp4

https://user-images.githubusercontent.com/12881364/200299337-6fe688dc-38a7-4feb-85a9-2243ab42ea20.mp4

## Issues Fixed
https://github.com/mobile-dev-inc/maestro/issues/251

### CODE REVIEW

File: maestro-client/src/main/java/maestro/drivers/IOSDriver.kt [hunk 1]
```diff
@@ -239,20 +243,70 @@ class IOSDriver(
...
+                    IdbIOSDevice.ScrollType.SWIPE
+                ).expect {}
+            }
+        }
+    }
+
     override fun backPress() {}
 
-    override fun hideKeyboard() {
+    override fun hideKeyboard() {3
```
Comment (Leland-Takamine) [inline maestro-client/src/main/java/maestro/drivers/IOSDriver.kt:309]: Typo

**Comment by Leland-Takamine** [comment]
What happens when direction and start/end are set? What about when only start is set but not end?

File: maestro-client/src/main/java/maestro/drivers/IOSDriver.kt (hunk 1 above)
Comment (amanjeetsingh150) [reply maestro-client/src/main/java/maestro/drivers/IOSDriver.kt]: Fixed ✅

**Comment by amanjeetsingh150** [comment]
> What happens when direction and start/end are set? What about when only start is set but not end?

Nice catch I have added a YamlSwipeDeserializer, which will look at fields. If cases like this happened I am throwing IllegalStateException:

```
java.lang.IllegalArgumentException: Provide swipe direction UP, DOWN, RIGHT, OR LEFT or by giving explicit start and end coordinates.
```

Let me know if you think the error message is not explicit enough or should be improved.

**Comment by amanjeetsingh150** [comment]
> I think the error message should be clearer

Hey @Leland-Takamine , have addressed the error messages. Have addressed all the common errors and raised different messages. Considered cases:

You can check error messages in this commit: https://github.com/mobile-dev-inc/maestro/pull/358/commits/d69476d37ffb1fd7992ee9c053597d038cf437bb

```
     1. Typo in direction command
     2. Typo in the start command
     3. Typo in the end command
     4. Providing direction start
     5. Providing direction end
     6. Providing direction start end
     7. Providing end only
     8. Providing start only
```

**Review by Leland-Takamine — APPROVED** [review]

**Pull request merged**